mongo/session.go

```diff
@@ -37,140 +56,50 @@ var withTransactionTimeout = 120 * time.Second
 // the provided callback. The other is to use NewSessionContext to explicitly create a SessionContext.
 type SessionContext interface {
 	context.Context
-	Session
```

Comment:
embedding a Session on a context is fairly confusing, as it allows users to use the context like a session.